<?xml version="1.0"?>
<launch>

	<param name="robot_description" command="$(find xacro)/xacro.py '$(find youbot_description)/robots/youbot_arm_only.urdf.xacro'"/>

     <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher">
        <param name="use_gui" value="true"/>
        <rosparam param="source_list">["poses/joint_states"]</rosparam>
    </node>

    <node name="dec2joint_translator_node" pkg="youbot_arm_control" type="dec2joint_translator.py" output="screen">
        <param name="file_with_trajectory_name" value="file_with_trajectory.txt"/>
        <rosparam param="min_q">[0.1, 0.1, -5.08, 0.1, 0.1]</rosparam>
        <rosparam param="max_q">[5.8, 2.6, -0.1, 3.48, 5.75]</rosparam>
    </node>

    <node name="trajectory_vizualizator_node" pkg="youbot_arm_control" type="trajectory_vizualizator.py" output="screen"/>

    <node name="decart_trajectory_publisher_node" pkg="youbot_arm_control" type="decart_trajectory_publisher.py" output="screen"/>

	<node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher"/>

    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find youbot_arm_control)/rviz/youbot_arm_model.rviz"/>

    <node pkg="tf" type="static_transform_publisher" name="ground_link_publisher" args="-0.024 0.0 0.032 0.0 0.0 0.0 ground_link base_link 100"/>

</launch>
